### PR TITLE
Run code formatter on GenServer

### PR DESCRIPTION
**Pull request by bschaeffer**
Couple of places where I don't understand the formatter output...

### CODE REVIEW

File: lib/elixir/lib/gen_server.ex [hunk 1]
```diff
@@ -479,8 +485,8 @@ defmodule GenServer do
   If `reason` is not `:normal`, `:shutdown`, nor `{:shutdown, term}` an error is
   logged.
   """
-  @callback terminate(reason, state :: term) ::
-    term when reason: :normal | :shutdown | {:shutdown, term} | term
+  @callback terminate(reason, state :: term) :: term
+            when reason: :normal | :shutdown | {:shutdown, term} | term
```
Comment (bschaeffer) [inline lib/elixir/lib/gen_server.ex]: Don't understand why the formatter moves `term` to the previous line... visually this is weird. If we have to expand these lines, I would expect the following to be valid.

```elixir
@callback terminate(reason, state :: term) ::
            term when reason: :normal
            | :shutdown
            | {:shutdown, term}
```

File: lib/elixir/lib/gen_server.ex [hunk 2]
```diff
@@ -479,8 +485,8 @@ defmodule GenServer do
   If `reason` is not `:normal`, `:shutdown`, nor `{:shutdown, term}` an error is
   logged.
   """
-  @callback terminate(reason, state :: term) ::
-    term when reason: :normal | :shutdown | {:shutdown, term} | term
+  @callback terminate(reason, state :: term) :: term
+            when reason: :normal | :shutdown | {:shutdown, term}
```
Comment (bschaeffer) [inline lib/elixir/lib/gen_server.ex]: Don't understand why the formatter moves `term` to the previous line... visually this is weird. If we have to expand these lines, I would expect the following to pass the formatter:

```elixir
@callback terminate(reason, state :: term) ::
            term when reason: :normal
            | :shutdown
            | {:shutdown, term}
```

But it get's converted to:

```elixir
@callback terminate(reason, state :: term) :: term
            when reason: :normal
                         | :shutdown
                         | {:shutdown, term}
```

File: lib/elixir/lib/gen_server.ex [hunk 3]
```diff
@@ -583,41 +592,48 @@ defmodule GenServer do
...
         proc =
           case Process.info(self(), :registered_name) do
-            {_, []}   -> self()
+            {_, []} -> self()
             {_, name} -> name
           end
-        :error_logger.error_msg('~p ~p received unexpected message in handle_info/2: ~p~n',
-                                [__MODULE__, proc, msg])
+
+        pattern = '~p ~p received unexpected message in handle_info/2: ~p~n'
```
Comment (bschaeffer) [inline lib/elixir/lib/gen_server.ex]: Could/should this be stored in a module attribute?

Comment (josevalim) [reply lib/elixir/lib/gen_server.ex]: It could but this way is completely fine.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**